### PR TITLE
moved Calib objects to DataFormats; Reimplement clusters as obj. with…

### PR DESCRIPTION
**Pull request by peressounko**
- moved Calib objects to DataFormats; 
- Reimplement clusters as obj. with cluster properties+ref to separate list of CluElements used to (re-)calculate E, position etc. 
- Re-implemented cluster unfolding procedure and turn it on by default

### CODE REVIEW

**Comment by peressounko** [comment]
Dear @shahor02, finally PR can be reviewed.

File: DataFormats/Detectors/PHOS/include/DataFormatsPHOS/BadChannelMap.h [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef O2_PHOS_PHOSBADCHANNELSMAP_H_
+#define O2_PHOS_PHOSBADCHANNELSMAP_H_
+
+#include <iosfwd>
+#include <bitset>
+#include <Rtypes.h>
+#include <CCDB/TObjectWrapper.h> // Needed to trigger dictionary build
```
Comment (shahor02) [inline DataFormats/Detectors/PHOS/include/DataFormatsPHOS/BadChannelMap.h]: why do you need this header here?

File: DataFormats/Detectors/PHOS/include/DataFormatsPHOS/Cluster.h [hunk 2]
```diff
@@ -21,6 +21,19 @@ class Geometry;
...
 
+struct CluElement {
+  short absId;
+  bool isHG;
+  int label;
+  float energy;
+  float time;
+  float localX;
+  float localZ;
+  float fraction;
```
Comment (shahor02) [inline DataFormats/Detectors/PHOS/include/DataFormatsPHOS/Cluster.h]: if you rely on default c-tor, better provide default init. values of data members, e.g. `short absId = 0;`

File: Detectors/PHOS/base/include/PHOSBase/Geometry.h [hunk 3]
```diff
@@ -129,6 +130,8 @@ class Geometry
   const TGeoHMatrix* getAlignmentMatrix(int mod) const { return &(mPHOS[mod]); }
 
  private:
+  static constexpr float cellStep = 2.25;
```
Comment (shahor02) [inline Detectors/PHOS/base/include/PHOSBase/Geometry.h]: constexpr members capitalized.

File: Detectors/PHOS/calib/src/PHOSBadMapCalibDevice.cxx [hunk 4]
```diff
@@ -253,26 +253,26 @@ void PHOSCalibCollector::writeOutputs()
...
-        mHistos[kTimeHGSlewing].Fill(ce->time, ce->energy);
-      } else {
-        if (ce->energy > mEminLGTime) {
-          mHistos[kTimeLGPerCell].Fill(absId, ce->time);
-        }
-        mHistos[kTimeLGSlewing].Fill(ce->time, ce->energy);
-      }
-    }
-  }
+  // // Fill time distributions only for cells in cluster
```
Comment (shahor02) [inline Detectors/PHOS/calib/src/PHOSBadMapCalibDevice.cxx:258]: if commented code is obsolete, please remove it

File: Detectors/PHOS/reconstruction/include/PHOSReconstruction/Clusterer.h [hunk 5]
```diff
@@ -81,18 +71,50 @@ class Clusterer
...
+  std::vector<float> mProp;             ///< proportion of clusters in the current digit
+  std::array<float, NLOCMAX> mxMax;     ///< current maximum coordinate
+  std::array<float, NLOCMAX> mzMax;     ///< in the unfolding procedure
+  std::array<float, NLOCMAX> meMax;     ///< currecnt amplitude in unfoding
+  std::array<float, NLOCMAX> mxMaxPrev; ///< coordunates at previous step
+  std::array<float, NLOCMAX> mzMaxPrev; ///< coordunates at previous step
+  std::array<float, NLOCMAX> dx;        ///< step on current minimization iteration
+  std::array<float, NLOCMAX> dz;        ///< step on current minimization iteration
+  std::array<float, NLOCMAX> dxprev;    ///< step on previoud minimization iteration
+  std::array<float, NLOCMAX> dzprev;    ///< step on previoud minimization iteration
```
Comment (shahor02) [inline Detectors/PHOS/reconstruction/include/PHOSReconstruction/Clusterer.h]: class data members names...

File: DataFormats/Detectors/PHOS/include/DataFormatsPHOS/BadChannelMap.h (hunk 1 above)
Comment (peressounko) [reply DataFormats/Detectors/PHOS/include/DataFormatsPHOS/BadChannelMap.h]: removed

File: DataFormats/Detectors/PHOS/include/DataFormatsPHOS/Cluster.h (hunk 2 above)
Comment (peressounko) [reply DataFormats/Detectors/PHOS/include/DataFormatsPHOS/Cluster.h]: added def values

File: Detectors/PHOS/base/include/PHOSBase/Geometry.h (hunk 3 above)
Comment (peressounko) [reply Detectors/PHOS/base/include/PHOSBase/Geometry.h]: fixed

File: Detectors/PHOS/calib/src/PHOSBadMapCalibDevice.cxx (hunk 4 above)
Comment (peressounko) [reply Detectors/PHOS/calib/src/PHOSBadMapCalibDevice.cxx:258]: This class still should be implemented. This is seed for final class

File: Detectors/PHOS/reconstruction/include/PHOSReconstruction/Clusterer.h (hunk 5 above)
Comment (peressounko) [reply Detectors/PHOS/reconstruction/include/PHOSReconstruction/Clusterer.h]: fixed

**Comment by peressounko** [comment]
Merge conflicts: files moved to another dir. How can I find what exactly conflicts? How to merge them?

**Comment by peressounko** [comment]
Check errors not related to PHOS/this commit

**Review by shahor02 — APPROVED** [review]

**Comment by shahor02** [comment]
FST fails for unrelated reasons (timeout), locally worked fine, merging.

**Pull request merged**